.github/workflows/main.yml

```diff
@@ -99,7 +99,7 @@ jobs:
 
   build-macos:
 
-    runs-on: macos-latest
+    runs-on: macos-14

```

Comment:
While 'latest' can help us detect issues earlier, I think pinning all of our CI tests to versions might be a better fit moving forward so we don't get random breakage when we're not ready to deal with it.